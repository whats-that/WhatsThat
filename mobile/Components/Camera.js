import React from "react";
import {
	Text,
	View,
	TouchableOpacity,
	ImageBackground,
	Button,
	CameraRoll,
	ScrollView,
	Image,
	ActivityIndicator
} from "react-native";
import { Camera, Permissions, FileSystem } from "expo";
import { Ionicons } from "@expo/vector-icons";
import axios from "axios";

export default class CameraView extends React.Component {
	constructor() {
		super();
		this.state = {
			hasCameraPermission: null,
			type: Camera.Constants.Type.back,
			newPhotos: false,
			previewImage: false,
			previewSource: "",
			photoBlob: {},
			isShowingPicture: true,
			cameraPhotos: [],
			cameraPhotosAreLoading: false
		};
		this.takePicture = this.takePicture.bind(this);
		this.onPictureSaved = this.onPictureSaved.bind(this);
		this.usePicture = this.usePicture.bind(this);
		this.getPhotos = this.getPhotos.bind(this);
		this.selectedPictureURI = this.selectedPictureURI.bind(this);
		this.closeCameraRoll = this.closeCameraRoll.bind(this);
	}

	selectedPictureURI(pictureURI) {
		this.setState({
			previewSource: pictureURI,
			previewImage: true,
			isShowingPicture: false
		});
	}

	closeCameraRoll() {
		this.setState({
			isShowingPicture: false
		});
	}

	async getPhotos() {
		this.setState({
			cameraPhotosAreLoading: true
		});

		try {
			const photos = await CameraRoll.getPhotos({
				first: 100,
				assetType: "All"
			});

			await this.setState({
				cameraPhotos: photos.edges,
				previewImage: false,
				isShowingPicture: true,
				cameraPhotosAreLoading: false
			});
		} catch (err) {
			console.warn("error loading images", err);
			this.setState({
				cameraPhotosAreLoading: false
			});
		}
	}

	async componentWillMount() {
		const { status } = await Permissions.askAsync(Permissions.CAMERA);
		this.setState({ hasCameraPermission: status === "granted" });
		FileSystem.makeDirectoryAsync(
			FileSystem.documentDirectory + "photos"
		).catch(e => {
			console.log(e, "Directory exists");
		});
	}

	async takePicture() {
		console.log("structural console.log; don't remove");
		if (this.camera) {
			this.camera.takePictureAsync({
				onPictureSaved: this.onPictureSaved,
				base64: true,
				quality: 0.1
			});
		}
	}

	onPictureSaved = async photo => {
		let destinationUri = `photos/${Date.now()}.jpg`;
		const destinationPath = `${FileSystem.documentDirectory}${destinationUri}`;
		destinationUri = "/" + destinationPath.substring(8);
		await FileSystem.moveAsync({
			from: photo.uri,
			to: destinationPath
		});
		await this.setState({
			previewImage: true,
			previewSource: destinationUri,
			newPhotos: true,
			photoBlob: photo
		});
	};
	async usePicture() {
<<<<<<< HEAD
<<<<<<< HEAD
		const result = await axios.post('http://whatsthat-capstone.herokuapp.com/api/server/getDataFromGoogleAPI', this.state.photoBlob)
		console.warn(result)
		return;
=======
		const result = await axios.post('http://172.16.23.112:8080/api/server/getDataFromGoogleAPI', this.state.photoBlob)
		console.log(result.data[0].webDetection.bestGuessLabels[0].label)

>>>>>>> b229a73fca886521b01133c02480f28810462f2d
=======
		// await axios.post('http://whatsthat-capstone.herokuapp.com/api/server', this.state.photoBlob)
		console.log(this.state.photoBlob);
		const result = await axios.post(
			"http://172.16.23.112:8080/api/server/getDataFromGoogleAPI",
			this.state.photoBlob
		);
		// console.log(result.data[0].webDetection.bestGuessLabels[0].label)
		// 	navigator.geolocation.getCurrentPosition(
		// 		(position) => {
		// 			 console.log(position)
		// 		},
		// 		(error) => alert(error.message),
		// 		{ enableHighAccuracy: true, timeout: 20000, maximumAge: 1000 }
		//  )
		return;
>>>>>>> 0e4616fd0b7bc395e32e3d9988502c5e98c912af
	}
	render() {
		const { hasCameraPermission, isShowingPicture } = this.state;
		if (this.state.cameraPhotosAreLoading) {
			return <ActivityIndicator style={{ width: "100%", height: "100%" }} />;
		}

		if (this.state.cameraPhotos.length > 0 && isShowingPicture) {
			return (
				<View>
					<ScrollView>
						{this.state.cameraPhotos.map((photo, i) => {
							return (
								<TouchableOpacity
									onPress={() => this.selectedPictureURI(photo.node.image.uri)}
									key={i}
								>
									<Image
										source={{ uri: photo.node.image.uri }}
										key={i}
										style={{ width: "100%", height: 300 }}
										resizeMode="cover"
										key={i}
									/>
								</TouchableOpacity>
							);
						})}
					</ScrollView>
					<TouchableOpacity
						onPress={this.closeCameraRoll}
						style={{ position: "absolute", paddingLeft: 10, bottom: 50 }}
					>
						<Ionicons name="md-close-circle" size={60} color="#cc0000" />
					</TouchableOpacity>
				</View>
			);
		}
		if (hasCameraPermission === null) {
			return <View />;
		} else if (hasCameraPermission === false) {
			return <Text>No access to camera</Text>;
		} else {
			if (this.state.previewImage) {
				return (
					<View style={{ flex: 1, bottom: 49 }}>
						<ImageBackground
							source={{ uri: this.state.previewSource }}
							resizeMode="cover"
							style={{ flex: 1, width: undefined, height: undefined }}
						>
							<View
								style={{
									position: "absolute",
									right: 0,
									bottom: 0,
									left: 0,
									backgroundColor: "transparent",
									flex: 1,
									flexDirection: "row",
									justifyContent: "space-between"
								}}
							>
								<TouchableOpacity
									onPress={() => this.setState({ previewImage: false })}
									style={{ alignSelf: "flex-end", paddingLeft: 10 }}
								>
									<Ionicons name="md-close-circle" size={60} color="#cc0000" />
								</TouchableOpacity>
								<TouchableOpacity
									onPress={this.usePicture}
									style={{ alignSelf: "flex-end", paddingRight: 10 }}
								>
									<Ionicons
										name="ios-arrow-dropright-circle"
										size={60}
										color="#00ffcc"
									/>
								</TouchableOpacity>
							</View>
						</ImageBackground>
					</View>
				);
			} else {
				return (
					<View style={{ flex: 1, bottom: 49 }}>
						<Camera
							style={{ flex: 1 }}
							type={this.state.type}
							ref={ref => {
								this.camera = ref;
							}}
						>
							<View
								style={{
									flex: 1,
									backgroundColor: "transparent",
									flexDirection: "row"
								}}
							/>
							<View
								style={{
									flex: 1,
									backgroundColor: "transparent",
									flexDirection: "row",
									alignSelf: "center"
								}}
							>
								<TouchableOpacity
									onPress={this.takePicture}
									style={{ alignSelf: "flex-end" }}
								>
									<Ionicons
										name="ios-radio-button-on"
										size={70}
										color="white"
									/>
								</TouchableOpacity>
								</View>
									<TouchableOpacity
										onPress={this.getPhotos}
										style={{ position: "absolute", bottom: 15, right: 15 }}
									>
										<Ionicons name="ios-photos" size={30} color="white" />
									</TouchableOpacity>
						</Camera>
					</View>
				);
			}
		}
	}
}
