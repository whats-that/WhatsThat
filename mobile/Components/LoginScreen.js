import React from "react";
import {
	Text,
	View,
	TouchableOpacity,
	ImageBackground,
	Button,
	TextInput
} from "react-native";

import { Camera, Permissions, FileSystem } from "expo";
import { Ionicons } from "@expo/vector-icons";

export default class LoginScreen extends React.Component {
	constructor() {
		super();
		this.state = {
			username: "",
			password: "",
			createAccount: false
		};
	}

	handleSubmit() {}

	render() {
		return (
			<ImageBackground
				style={{ flex: 1 }}
				source={{
					uri:
						"https://cdn-images-1.medium.com/max/1920/1*AcYLHh0_ve4TNRi6HLFcPA.jpeg"
				}}
			>
				<View
					style={{
						backgroundColor: "transparent",
						flex: 1,
						flexDirection: "column",
						justifyContent: "center",
						alignItems: "center"
					}}
				>
					<Text style={{ fontSize: 32, color: "white", textAlign: "center" }}>
						Welcome to What's That!
					</Text>
					{this.state.createAccount ? (
						<Text style={{ fontSize: 20, color: "white", textAlign: "center" }}>
							Please create an account to continue
						</Text>
					) : (
						<Text style={{ fontSize: 20, color: "white", textAlign: "center" }}>
							Please sign in to continue
						</Text>
					)}
					<View
						style={{
							justifyContent: "center",
							marginTop: 30,
							backgroundColor: "transparent"
						}}
					>
						<TextInput
							style={{ height: 40, width: 300, textAlign: "center" }}
							placeholder="Username"
							onChangeText={text => this.setState({ username: text })}
							value={this.state.username}
						/>
						<TextInput
							style={{ height: 40, width: 300, textAlign: "center" }}
							placeholder="Password"
							secureTextEntry={true}
							onChangeText={text => this.setState({ password: text })}
							value={this.state.password}
						/>
						{!this.state.createAccount ? (
							<Button
								onPress={() => this.setState({ createAccount: true })}
								style={{}}
								title="Create Account"
							/>
						) : (
							<View />
						)}
						{this.state.password && this.state.username ? (
							<Button onPress={this.handleSubmit} style={{}} title="Submit" />
						) : (
							<View />
						)}
						{this.state.createAccount ? (
							<TextInput
								style={{ height: 40, width: 300, textAlign: "center" }}
								placeholder="Confirm Password"
								secureTextEntry={true}
								onChangeText={text => this.setState({ password: text })}
								value={this.state.password}
							/>
						) : (
							<View />
						)}
					</View>
				</View>
			</ImageBackground>
		);
	}
}
