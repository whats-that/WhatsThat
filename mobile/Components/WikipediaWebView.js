import React from 'react';
import {WebView} from 'react-native';

export default class WikipediaWebView extends React.Component {
    constructor(){
        super();

        this.suffix = 'rocky statue';

        this.replaceSpacesWithUnderScores = this.replaceSpacesWithUnderScores.bind(this);
    }

    replaceSpacesWithUnderScores(){
        const newString = this.suffix.split(' ').join('_');
        return newString;
    }

    render(){

        return (
            <WebView
            source={{uri: 'https://www.wikipedia.org/wiki/' + this.replaceSpacesWithUnderScores()}} />
        );
    }
}
