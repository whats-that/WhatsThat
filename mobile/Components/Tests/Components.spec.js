import Camera from '../Camera';
import LoginScreen from '../LoginScreen';
// import PastLandMarks from '../PastLandMarks';
// import SingleLandMark from '../SingleLandMark';
import WikipediaWebView from '../WikipediaWebView';

import React from 'react';
import renderer from 'react-test-renderer';


test('Camera renders correctly', () => {
  const tree = renderer.create(<Camera />).toJSON();
  expect(tree).toMatchSnapshot();
});

test('Login Screen renders correctly', () => {
  const tree = renderer.create(<LoginScreen />).toJSON();
  expect(tree).toMatchSnapshot();
});

// test('Past Landmarks view renders correctly', () => {
//   const tree = renderer.create(<PastLandMarks />).toJSON();
//   expect(tree).toMatchSnapshot();
// });

// test('Single Landmark view renders correctly', () => {
//   const tree = renderer.create(<SingleLandMark />).toJSON();
//   expect(tree).toMatchSnapshot();
// });

test('Wikipedia view renders correctly', () => {
  const tree = renderer.create(<WikipediaWebView />).toJSON();
  expect(tree).toMatchSnapshot();
});