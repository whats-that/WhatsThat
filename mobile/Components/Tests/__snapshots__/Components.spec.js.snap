// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Camera renders correctly 1`] = `<View />`;

exports[`Login Screen renders correctly 1`] = `
<View
  style={
    Object {
      "flex": 1,
    }
  }
>
  <Image
    source={
      Object {
        "uri": "https://cdn-images-1.medium.com/max/1920/1*AcYLHh0_ve4TNRi6HLFcPA.jpeg",
      }
    }
    style={
      Array [
        Object {
          "bottom": 0,
          "left": 0,
          "position": "absolute",
          "right": 0,
          "top": 0,
        },
        Object {
          "height": undefined,
          "width": undefined,
        },
        undefined,
      ]
    }
  />
  <View
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "transparent",
        "flex": 1,
        "flexDirection": "column",
        "justifyContent": "center",
      }
    }
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "color": "white",
          "fontSize": 32,
          "textAlign": "center",
        }
      }
    >
      Welcome to What's That!
    </Text>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "color": "white",
          "fontSize": 20,
          "textAlign": "center",
        }
      }
    >
      Please sign in to continue
    </Text>
    <View
      style={
        Object {
          "backgroundColor": "transparent",
          "justifyContent": "center",
          "marginTop": 30,
        }
      }
    >
      <TextInput
        allowFontScaling={true}
        onChangeText={[Function]}
        placeholder="Username"
        style={
          Object {
            "height": 40,
            "textAlign": "center",
            "width": 300,
          }
        }
        value=""
      />
      <TextInput
        allowFontScaling={true}
        onChangeText={[Function]}
        placeholder="Password"
        secureTextEntry={true}
        style={
          Object {
            "height": 40,
            "textAlign": "center",
            "width": 300,
          }
        }
        value=""
      />
      <View
        accessibilityComponentType="button"
        accessibilityTraits={
          Array [
            "button",
          ]
        }
        accessible={true}
        isTVSelectable={true}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "opacity": 1,
          }
        }
      >
        <View
          style={
            Array [
              Object {},
            ]
          }
        >
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Array [
                Object {
                  "color": "#007AFF",
                  "fontSize": 18,
                  "padding": 8,
                  "textAlign": "center",
                },
              ]
            }
          >
            Create Account
          </Text>
        </View>
      </View>
      <View />
      <View />
      <View
        accessibilityComponentType="button"
        accessibilityTraits={
          Array [
            "button",
          ]
        }
        accessible={true}
        isTVSelectable={true}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "opacity": 1,
          }
        }
      >
        <View
          style={
            Array [
              Object {},
            ]
          }
        >
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Array [
                Object {
                  "color": "#007AFF",
                  "fontSize": 18,
                  "padding": 8,
                  "textAlign": "center",
                },
              ]
            }
          >
            Submit
          </Text>
        </View>
      </View>
    </View>
  </View>
</View>
`;

exports[`Wikipedia view renders correctly 1`] = `
<View
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View
    style={
      Object {
        "flex": 1,
      }
    }
  >
    <RCTWebView
      messagingEnabled={false}
      onLoadingError={[Function]}
      onLoadingFinish={[Function]}
      onLoadingStart={[Function]}
      onMessage={[Function]}
      scalesPageToFit={true}
      source={
        Object {
          "uri": "https://www.wikipedia.org/wiki/Empire State Building",
        }
      }
      style={
        Array [
          Object {
            "flex": 1,
          },
          Object {
            "backgroundColor": "#ffffff",
          },
          undefined,
        ]
      }
    />
  </View>
  <ActivityIndicator
    animating={true}
    color="#999999"
    hidesWhenStopped={true}
    size="small"
    style={
      Object {
        "height": "100%",
        "width": "100%",
      }
    }
  />
</View>
`;
